app/assets/javascripts/template_invocation.js

```diff
@@ -24,6 +24,33 @@ function refresh_search_query(value){
   $('textarea#targeting_search_query').val($('span#bookmark_query_map span#bookmark-' + id).data('query'));
 }
 
+function show_preview_hosts_modal() {
+  var modal_window = $('#previewHostsModal');
+
+  var form = $('form#job_invocation_form');
+  var data = form.serializeArray();
+
+  request = $.ajax({
+    data: data,
+    type: 'POST',
+    url: modal_window.attr('data-url'),
+    success: function(request) {
+      modal_window.find('.modal-body').html(request);
```

Comment:
would it make sense to use `#previewHostsModal .modal-body` just in case there are more modal windows on the page?